Clinical-Genomics/scout
# Updated Loqusdb table to display information on if current case is the one counted when the observations equals one

This PR adds a functionality and fixes a bug.

The changes in this PR implements a check which determines if the current case is included in the observations of a specific variant from a Loqusdb instance. When observations=1 the information is used to choose which text and tooltip to display (Observed only in this case / Observed only in another case). The text is displayed regardless of if any cases from the [families] registered in Loqusdb are found in the scout-database. 

These changes was intended as a fix for https://github.com/Clinical-Genomics/scout/issues/4384, the changes however does not fully solve the issue. But it closes https://github.com/Clinical-Genomics/scout/issues/3469 since the same-case verification is now implemented when observations=1.



<details>
<summary>Testing on cg-vm1 server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. Make sure the PR is pushed and available on [Docker Hub](https://hub.docker.com/repository/docker/clinicalgenomics/scout-server-stage)
1. Fist book your testing time using the Pax software available at [https://pax.scilifelab.se/](https://pax.scilifelab.se). The resource you are going to call dibs on is `scout-stage` and the server is `cg-vm1`.
1. `ssh <USER.NAME>@cg-vm1.scilifelab.se`
1. `sudo -iu hiseq.clinical`
1. `ssh localhost`
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `podman ps`
1. Stop the service with current deployed branch: `systemctl --user stop scout.target`
1. Start the scout service with the branch to test: `systemctl --user start scout@<this_branch>`
1. Make sure the branch is deployed: `systemctl --user status scout.target`
1. After testing is done, repeat procedure at [https://pax.scilifelab.se/](https://pax.scilifelab.se), which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>

<details>
<summary>Testing on hasta server (Clinical Genomics Stockholm)</summary>

**Prepare for testing**
1. `ssh <USER.NAME>@hasta.scilifelab.se`
1. Book your testing time using the Pax software. `us; paxa -u <user> -s hasta -r scout-stage`. You can also use the WSGI Pax app available at [https://pax.scilifelab.se/](https://pax.scilifelab.se).
1. (optional) Find out which scout branch is currently deployed on cg-vm1: `conda activate S_scout; pip freeze | grep scout-browser`
1. Deploy the branch to test: `bash /home/proj/production/servers/resources/hasta.scilifelab.se/update-tool-stage.sh -e S_scout -t scout -b <this_branch>`
1. Make sure the branch is deployed: `us; scout --version`
1. After testing is done, repeat the `paxa` procedure, which will release the allocated resource (`scout-stage`) to be used for testing by other users.
</details>


**How to test**:
1. Identify suitable test-variants which ideally have one observation in at least one of the Louqusdb instances. One where the case_id from scout is present in the [families] of the Loqusdb observations for the variant and one where it is not.
2. Navigate to the variant page in scout. The tr with information on if the observation is from the same or another case should be visible and the text-content correct. If the observation was made for another case and the user has access to the case it is displayed next to the text. If the case was not found in scout or the user does not have access no link will be present. 
3. Check the tooltips which have been added. Same-case and other-case observations have differing tooltip contents. 
4. The main tooltip for the card referring to greyed out links was moved to the actual missing links.

**Expected outcome**:
The functionality should be working
Take a screenshot and attach or copy/paste the output.

**Review:**
